src/helpers/serverModuleHelpers.js

```diff
@@ -0,0 +1,122 @@
+const { existsSync } = require('node:fs')
+const { readFile, writeFile, rename, rm } = require('node:fs/promises')
+const { parse, join } = require('node:path')
+
+const { satisfies } = require('semver')
+
+const getAngularJson = require('./getAngularJson')
+const { getProject } = require('./setUpEdgeFunction')
+
+// eslint-disable-next-line no-inline-comments
+const NetlifyServerTsCommonEngine = /* typescript */ `import { CommonEngine } from '@angular/ssr/node'
+import { render } from '@netlify/angular-runtime/common-engine'
+
+const commonEngine = new CommonEngine()
+
+export default async function HttpHandler(request: Request, context: any): Promise<Response> {
+  return await render(commonEngine)
+}
+`
+
+// eslint-disable-next-line no-inline-comments
+const NetlifyServerTsAppEngine = /* typescript */ `import { AngularAppEngine, createRequestHandler } from '@angular/ssr'
+const angularAppEngine = new AngularAppEngine()
+
+// @ts-expect-error - createRequestHandler expects a function with single Request argument and doesn't allow context argument
+export const reqHandler = createRequestHandler(async (request: Request, context: any) => {
+  const result = await angularAppEngine.handle(request, context)
+  return result || new Response('Not found', { status: 404 })
+})
+`
+
+let needSwapping = false
+let serverModuleLocation
+let serverModuleBackupLocation
```

Comment:
😬 hrm, module scope state? I guess since this is all running as part of a build plugin it can ever run twice without a process lifecycle, so it's probably ok. If it's easy I'd suggest refactoring it away though, since it's harder to reason about and introduces potential for future bugs and test nondeterminism.

Reply:
Not super happy about this, but I just followed with the way we held state between build hooks already ( https://github.com/netlify/angular-runtime/blob/main/src/index.js#L8 ). I also don't really see good way to handle it that doesn't ultimately rely on holding state in module or global scope (that might just be abstracted away at first glance)